src/Pickles/Pickles.TestFrameworks/VsTest/VsTestScenarioOutlineExampleMatcher.cs

```diff
@@ -18,30 +18,173 @@
 //  </copyright>
 //  --------------------------------------------------------------------------------------------------------------------
 
+using System.Collections.Generic;
 using System.Linq;
 using System.Text.RegularExpressions;
 using System.Xml.Linq;
-
 using PicklesDoc.Pickles.ObjectModel;
 
 namespace PicklesDoc.Pickles.TestFrameworks.VsTest
 {
     public class VsTestScenarioOutlineExampleMatcher : IScenarioOutlineExampleMatcher
     {
-        private static readonly Regex VariantRegex = new Regex(@"(.*)_Variant([\d*])", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        private const int VariantNumberGroup = 2;
+        private static readonly Regex VariantWithExampleGroupRegex = new Regex(@"(?:[\S]+)_ExampleSet([\d]+)_Variant([\d]+)$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
This is different than the regex I initially created, I'll have a look tomorrow to see if this works for me as well.

Reply:
It does give me some different behavior, I will check if I can come up with a test file that exposes these issues.